data_node/controllers/outer/job_execution_controller.go

```diff
@@ -0,0 +1,52 @@
+package outer
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+
+	"github.com/SayedAlesawy/Videra-Storage/config"
+	datanode "github.com/SayedAlesawy/Videra-Storage/data_node"
+	"github.com/SayedAlesawy/Videra-Storage/utils/errors"
+)
+
+var jobExecutionLoggerPrefix = "[Job-Execution]"
+
+func startJob(videoInfo datanode.File) {
+
+	var metadata datanode.VideoMetadata
+	json.Unmarshal([]byte(videoInfo.Extras), &metadata)
+
+	var modelInfo datanode.File
+	datanode.NodeInstance().DB.Connection.Where("token = ?", metadata.AssociatedModel).Find(&modelInfo)
+
+	var modelExtras datanode.ModelExtras
+	json.Unmarshal([]byte(modelInfo.Extras), &modelExtras)
+
+	executeJob(videoInfo.Path, modelInfo.Path, modelExtras.AssociatedConfigPath, modelExtras.AssociatedCodePath, modelInfo.Token, 0, metadata.FramesCount)
+}
+
+func executeJob(videoPath string, modelPath string, configPath string, codePath string, groupID string, startIndex int, framesCount int) {
+	command := config.ConfigurationManagerInstance("").DataNodeConfig().IngestionModuleCommand
+	args := prepareArgs(videoPath, modelPath, configPath, codePath, groupID, startIndex, framesCount)
+	cmd := exec.Command(command, args)
+	cmd.Dir = config.ConfigurationManagerInstance("").DataNodeConfig().IngestionModulePath
+	err := cmd.Run()
+	if errors.IsError(err) {
+		log.Println(jobExecutionLoggerPrefix, err)
```

Comment:
I think it might be useful to return the error back to the sdk as a response of the last chunk upload. This way we can do re-tries from the sdk side. But we need to make sure that the last chunk is written if the job start fails.